javascript/extractor/src/com/semmle/js/extractor/AutoBuild.java

```diff
@@ -1238,18 +1241,24 @@ private void doExtract(FileExtractor extractor, Path file, ExtractorState state)
       for (ParseError err : errors) {
         String msg = "A parse error occurred: " + StringUtil.quoteWithBackticks(err.getMessage().trim())
             + ". Check the syntax of the file. If the file is invalid, correct the error or [exclude](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/customizing-code-scanning) the file from analysis.";
-        // file, relative to the source root
-        DiagnosticLocation.Builder builder = DiagnosticLocation.builder();
+
+        Optional<DiagnosticLocation> diagLoc = Optional.empty();
         if (file.startsWith(LGTM_SRC)) {
-          builder = builder.setFile(file.subpath(LGTM_SRC.getNameCount(), file.getNameCount()).toString());
-        }
-        DiagnosticLocation diagLoc = builder
+          diagLoc = DiagnosticLocation.builder()
+            .setFile(file.subpath(LGTM_SRC.getNameCount(), file.getNameCount()).toString()) // file, relative to the source root
             .setStartLine(err.getPosition().getLine())
             .setStartColumn(err.getPosition().getColumn() + 1) // convert from 0-based to 1-based
             .setEndLine(err.getPosition().getLine())
             .setEndColumn(err.getPosition().getColumn() + 1) // convert from 0-based to 1-based
-            .build();
-        writeDiagnostics(msg, JSDiagnosticKind.PARSE_ERROR, diagLoc);
+            .build()
+            .getOk();
+        }
+        if (diagLoc.isPresent()) {
+          writeDiagnostics(msg, JSDiagnosticKind.PARSE_ERROR, diagLoc.get());
+        } else {
+          msg += "\n\nRelated file: " + file;
```

Comment:
I'm not sure we should "leak" the folder hierarchy of the machine on which the analysis was run into the diagnostic message. 

We should probably include a `DiagnosticLocation` with just the file path if the `build()` method fails due to invalid position information.

Reply:
> I'm not sure we should "leak" the folder hierarchy of the machine on which the analysis was run into the diagnostic message.

That seems reasonable, particularly for the status page.  Done.

> We should probably include a `DiagnosticLocation` with just the file path if the `build()` method fails due to invalid position information.

We're just checking startLine <= endLine, so my understanding is we should never fail here due to invalid position information.